integration/oasis/reachable_by_multiple_charge.go

```diff
@@ -0,0 +1,79 @@
+package oasis
+
+import (
+	"github.com/Telenav/osrm-backend/integration/oasis/haversine"
+	"github.com/Telenav/osrm-backend/integration/oasis/stationfinder"
+	"github.com/Telenav/osrm-backend/integration/pkg/api/oasis"
+	"github.com/Telenav/osrm-backend/integration/pkg/api/osrm/route"
+	"github.com/golang/glog"
+
+	"github.com/twpayne/go-polyline"
+)
+
+// For each route response, will generate an array of *stationfinder.StationCoordinate
+// Each array contains: start point, first charge point(could also be start point), second charge point, ..., end point
+func chargeLocationSelection(oasisReq *oasis.Request, routeResp *route.Response) [][]*stationfinder.StationCoordinate {
+	results := [][]*stationfinder.StationCoordinate{}
+	for _, route := range routeResp.Routes {
+
+		result := []*stationfinder.StationCoordinate{}
+		result = append(result, &stationfinder.StationCoordinate{
+			Lat: oasisReq.Coordinates[0].Lat,
+			Lon: oasisReq.Coordinates[0].Lon})
+		currEnergy := oasisReq.CurrRange
+
+		// if initial energy is too low
+		if currEnergy < oasisReq.PreferLevel {
+			result = append(result, &stationfinder.StationCoordinate{
+				Lat: oasisReq.Coordinates[0].Lat,
+				Lon: oasisReq.Coordinates[0].Lon})
+			currEnergy = oasisReq.MaxRange
+		}
+
+		result, currEnergy = findChargeLocation4Route(route, result, currEnergy, oasisReq.PreferLevel, oasisReq.MaxRange)
+		if len(result) != 0 {
+			result = append(result, &stationfinder.StationCoordinate{
+				Lat: oasisReq.Coordinates[1].Lat,
+				Lon: oasisReq.Coordinates[1].Lon})
+			results = append(results, result)
+		}
+	}
+	return results
+}
+
+func findChargeLocation4Route(route *route.Route, result []*stationfinder.StationCoordinate, currEnergy, preferLevel, maxRange float64) ([]*stationfinder.StationCoordinate, float64) {
+	for _, leg := range route.Legs {
+		for _, step := range leg.Steps {
+			if (currEnergy - step.Distance) < preferLevel {
+				coords, _, err := polyline.DecodeCoords([]byte(step.Geometry))
+				if err != nil {
+					glog.Errorf("Incorrect geometry encoding string from route response, error=%v", err)
```

Comment:
Error strings should not be capitalized . See https://github.com/golang/go/wiki/CodeReviewComments#error-strings